### PR TITLE
feat: add FoxPage MainOpportunity component

### LINKED ISSUE #1785: Fox Page - Main Opportunity
### Overview

We should create a component to display the main opportunity for FOX/FOXy, with all its info and a CTA to get started with this opportunity.

### References and additional details

`FoxyOpportunity` can be used as a base type for this, using a subset of its fields.
Given we represent opportunities in a standardized way across the codebase, it is safe to assume such subset without API changes needed later on. From this type, pick these to be passed as props:

- `apy`
- `tvl`
- `balance` (the balance in tokens)
- `type` and `provider` will more than likely be needed (waiting on architecture planning)
- `tokenCaip19` - the caip19 of the staked token (can be different from the reward token), where the token`name` and description can be gotten from
  - Note: To be fully functional and be able to develop this in isolation, not using `tokenCaip19` and instead passing the name and description as separate props is probably a better way to go with this 

https://github.com/shapeshift/web/blob/fd67442e47da2136ddf7148fa3549c330a1b89e5/src/pages/Defi/hooks/useFoxyBalances.tsx#L19-L36

### Acceptance Criteria

- A `<asset symbol> Asset header` is displayed (hardcoded FOX asset for now)
- The description of that asset is displayed (hardcoded FOX asset for now)
- The APY is displayed as a percentage
- The TVL is displayed in Fiat with `US$` prefix similar to the  DeFi section
- The token balance is shown if there is a balance, `--` otherwise
- "Get Started" button is displayed but not yet clickable - it will open different steps depending on the staking type

Design note:  As we've discussed during the design requirements, all internal CTAs like this "Get Started" should be displayed with the full style:
image

While all external links are displayed as ghost buttons:
image

For this issue, assume FOX staking opening the FOXy modal, which is an internal CTA. 

### Need By Date

_No response_

### Screenshots/Mockups

Mockups: https://www.figma.com/proto/WgALgik9YuWzq7VIM0JRdb/FOX-Page-Bounty?node-id=71%3A533&scaling=min-zoom&page-id=71%3A532
image

### Estimated effort

3 hours

### PR DESCRIPTION
**Pull request by NeOMakinG**
## Description
I added a new component called `MainOpportunity`
It currently has a few props:
```ts
type MainOpportunityProps = {
  apy: string
  tvl: BigNumber
  balance: string
  onClick: () => void
}
```
This API will probably change in the future depending on how we design its functionality

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)
closes #1785

## Risk
No risk

## Testing
Check on storybook that the component looks like mockups

## Screenshots (if applicable)
image

### CODE REVIEW

**Review by elmutt — APPROVED** [review]

File: src/plugins/foxPage/components/Layout.stories.tsx [hunk 1]
```diff
@@ -80,7 +82,12 @@ export const FoxLayout: Story = () => {
               direction={{ base: 'column', xl: 'row' }}
             >
               <Stack spacing={4} flex='1 1 0%' width='full'>
-                <Text>{'Fox Page'}</Text>
+                <MainOpportunity
+                  balance='--'
+                  tvl={bnOrZero(150000000)}
+                  apy='11.61%'
```
Comment (gomesalexandre) [inline src/plugins/foxPage/components/Layout.stories.tsx:88]:
```suggestion
                  apy='0.1161'
```

File: src/plugins/foxPage/components/Layout.stories.tsx [hunk 2]
```diff
@@ -80,7 +82,12 @@ export const FoxLayout: Story = () => {
               direction={{ base: 'column', xl: 'row' }}
             >
               <Stack spacing={4} flex='1 1 0%' width='full'>
-                <Text>{'Fox Page'}</Text>
+                <MainOpportunity
+                  balance='--'
+                  tvl={bnOrZero(150000000)}
```
Comment (gomesalexandre) [inline src/plugins/foxPage/components/Layout.stories.tsx:87]: This should be a string as well, let the component make it a BN, as they don't serialize well.
```suggestion
                  tvl={'150000000'}
```

File: src/plugins/foxPage/components/MainOpportunity.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  const translate = useTranslate()
+
+  return (
+    <Card display='block' width='full' borderRadius={8}>
+      <Card.Header>
+        <Flex flexDirection='row' alignItems='center' mb={2}>
+          <AssetIcon src={fox.icon} boxSize='6' mr={2} zIndex={2} />
+          <CText fontWeight='bold' color='inherit'>
+            {translate('plugins.foxPage.titleStaking', {
+              assetSymbol: 'FOX',
```
Comment (gomesalexandre) [inline src/plugins/foxPage/components/MainOpportunity.tsx:28]: Pass this one from props also?

File: src/plugins/foxPage/components/MainOpportunity.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+              {apy}
+            </CText>
+          </Box>
+          <Box display={{ base: 'none', md: 'block' }} width={{ base: '50%', md: 'auto' }}>
+            <Text translation='plugins.foxPage.tvl' color='gray.500' mb={1} />
+            <Amount.Fiat
+              color='inherit'
+              fontSize={'xl'}
+              fontWeight='semibold'
+              value={tvl.toString()}
```
Comment (gomesalexandre) [inline src/plugins/foxPage/components/MainOpportunity.tsx:52]: If you don't BN it, you won't need to `.toString()` it here

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
A few changes requested WRT the API
Ran the Storybook locally and it LGTM visually, ready to stamp it after comments are tackled!

File: src/plugins/foxPage/components/MainOpportunity.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+import { BigNumber } from 'lib/bignumber/bignumber'
+
+type MainOpportunityProps = {
+  apy: string
+  tvl: BigNumber
+  balance: string
+  onClick: () => void
+}
+
+export const MainOpportunity = ({ apy, tvl, balance, onClick }: MainOpportunityProps) => {
```
Comment (theoboldfrazier) [inline src/plugins/foxPage/components/MainOpportunity.tsx:18]: I'm wondering if `MainOpportunity` is the best naming here. It's very ambiguous when we consider non-fox opportunities down the road.

Perhaps `FoxOpportunity` might be better to clarify that this is directly related to general Fox earning opportunities.

**Comment by NeOMakinG** [comment]
Thanks for the review, should be fine now :)

**Review by theoboldfrazier — APPROVED** [review]

File: src/plugins/foxPage/components/FoxOpportunity.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  onClick: () => void
+}
+
+export const FoxOpportunity = ({
+  apy,
+  tvl,
+  assetSymbol,
+  balance,
+  onClick,
+}: MainOpportunityProps) => {
```
Comment (0xdef1cafe) [inline src/plugins/foxPage/components/FoxOpportunity.tsx:24]:
```suggestion
export const FoxOpportunity: React.FC<MainOpportunityProps> = ({
  apy,
  tvl,
  assetSymbol,
  balance,
  onClick,
}) => {
```

non-blocking comment, but this is always the nicer way to do things - type the function, not the args

**Review by 0xdef1cafe — APPROVED** [review]

**Review by gomesalexandre — APPROVED** [review]
Pulled locally, story still LGTM. Component has been properly updated with an `Amount.Percent` to accomodate the new 0.-format APY.

**Pull request merged**